### PR TITLE
feat(dropdown/basic): allow handlers on items, and close list if clic…

### PR DESCRIPTION
**Pull request by ruben-martin-lozano**
In this PR we allow to:
- Item links to have a click handler
- Close list if document click is triggered
- Close list if item link click is triggered

To keep the current behaviour in all projects, default values to new features is set to false.

### CODE REVIEW

File: components/dropdown/basic/src/index.js [hunk 1]
```diff
@@ -70,14 +71,18 @@ class DropdownBasic extends Component {
...
-  _renderLink = ({text, url, target}, index) => {
+  _renderLink = ({onClick, target, text, url}, index) => {
     const Link = this.props.linkFactory
 
     return (
       <li key={index} className="sui-DropdownBasicMenu-listItem">
         <Link
           href={url}
           className="sui-DropdownBasicMenu-listLink"
+          onClick={e => {
```
Comment (carlosvillu) [inline components/dropdown/basic/src/index.js]: Don't create inline arrow functions to handle events. They result in more memory consuming.

File: components/dropdown/basic/src/index.js [hunk 2]
```diff
@@ -118,7 +151,10 @@ class DropdownBasic extends Component {
             <ArrowButtonIcon svgClass="sui-DropdownBasic-buttonIcon" />
           </button>
         </div>
-        <div className="sui-DropdownBasicMenu">
+        <div
+          className="sui-DropdownBasicMenu"
+          ref={node => (this.wrapper = node)}
```
Comment (carlosvillu) [inline components/dropdown/basic/src/index.js]: We should start to use the new Ref API more align with the next React version.

File: components/dropdown/basic/src/index.js (hunk 1 above)
Comment (ruben-martin-lozano) [reply components/dropdown/basic/src/index.js]: 👍

**Comment by ruben-martin-lozano** [comment]
@carlosvillu feedback aplicado ;)

**Pull request merged**